scripts/linters/pylint_extensions.py

```diff
@@ -1922,6 +1937,110 @@ def process_tokens(self, tokens):
                         'single-line-pragma', line=line_num)
 
 
+class TypeIgnoreCommentChecker(checkers.BaseChecker):
+    """Custom pylint checker which checks if MyPy's type ignores are properly
+    documented or not.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    EXCLUDED_DIRS_HAVING_IGNORE_TYPE_COMMENTS = (
+        EXCLUDED_TYPE_COMMENT_DIRECTORIES
+    )
+
+    name = 'type-ignore-comment'
+    priority = -1
+    msgs = {
+        'C0045': (
+            'Please try to avoid the use of type: ignore where possible.'
+            ' If \'type: ignore\' is really necessary, then add a proper'
+            ' comment with clear justification (see other parts of the'
+            ' codebase for examples).',
+            'mypy-ignore-used',
+            'MyPy ignores should be used with proper comments following'
+            ' the established patterns in the codebase. Except for'
+            ' no-untyped-call MyPy ignore.'
+        ),
+        'C0046': (
+            'Extra comment is present for MyPy type: ignore. Please'
+            ' remove it.',
+            'redundant-type-comment',
+            'No corresponding \'type: ignore\' is found for the comment.'
+        )
+    }
+
+    def visit_module(self, node):
+        """Visit a module to ensure that there is a comment for each MyPy
+        type ignore.
+
+        Args:
+            node: astroid.scoped_nodes.Module. Node to access module content.
+        """
+        for directory in self.EXCLUDED_DIRS_HAVING_IGNORE_TYPE_COMMENTS:
+            if directory in node.root().file:
+                return
+        tokens = pylint_utils.tokenize_module(node)
+        self._process_module_tokens(tokens, node)
+
+    def _process_module_tokens(self, tokens, node):
+        """Checks if the MyPy type ignores present in a module properly
+        documented by a code comment or not.
```

Comment:
```suggestion
        """Checks if the MyPy type ignores present in a module are properly
        documented by a code comment or not.
```

Also checks for unnecessary code comments

Reply:
Fixed -- wherever I found these issues.